app/questionnaire/questionnaire_store_updater.py

```diff
@@ -193,9 +193,9 @@ def remove_relationship_answers_for_list_item_id(
             self._answer_store.add_or_update(answer)
 
     def add_completed_location(self, location: Optional[Location] = None):
-        location = location or self._current_location
-
-        self._progress_store.add_completed_location(location)
+        if not self._progress_store.is_routing_backwards:
```

Comment:
Due to how our current architecture is structured the block was being removed, but because this came after routing it was immediate being put. Nearly send me nuts as unit tests confirmed it was being removed, but in UI it was still there, argggg.